node_modules/col-core/lib/db.js

```diff
@@ -894,6 +894,17 @@ var setUpModel = function(sequelize) {
   });
   Asset.hasMany(Activity);
 
+  // An activity can be associated to an comment, but we specify no foreign key because 'object_id' may point
+  // to things other than comments. When using this association in queries, specify a join on 'object_id'
+  // only when 'object_type' = 'comment'. In Sequelize idiom:
+  // 'on': {
+  //   '$activities.object_type$': 'comment',
+  //   '$activities.comment.id$': {$col: 'activities.object_id'}
+  // }
+  Activity.belongsTo(Comment, {
+    'constraints': 'false'
```

Comment:
@johncrossman Ha, no; that might have accounted for some other problems. Amending.